azurerm/internal/services/compute/virtual_machine_scale_set_data_source.go

```diff
@@ -31,6 +39,8 @@ func dataSourceVirtualMachineScaleSet() *schema.Resource {
 
 			"location": azure.SchemaLocationForDataSource(),
 
+			"network_profile": schemaNetworkProfileForDataSource(),
```

Comment:
which would make this
```suggestion
			"network_interface": VirtualMachineScaleSetNetworkInterfaceSchema(),
```